### ISSUE TITLE
Conflict between rules RF04 and RF06

### ISSUE THREAD

**Issue by craiga** [sqlfluff/sqlfluff#6385]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The rules RF04 and RF06 don't allow a column with the same name as a keyword, regardless of whether the column is quoted or not.

### Expected Behaviour

I'd expect to be allowed a column with the same name as a keyword, as long as the name of the column is quoted.

```sql
SELECT 1978 AS "year";
```
👍 

```sql
SELECT 1978 AS year;
```
👎 

### Observed Behaviour

Quoting the value results in a RF04 error. Unquoting the value results in a RF06.

### How to reproduce

Create a file `test.sql` which looks like this:

```sql
SELECT 1978 AS year;
```

Running lint on this in the Postgres dialect flags that you're using the year keyword as an identifier.

```
❯ sqlfluff lint --dialect postgres test.sql
== [test.sql] FAIL                                                                                                                                                        
L:   1 | P:  16 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
All Finished 📜 🎉!
```

Fair enough. Update `test.sql`.

```sql
SELECT 1978 AS "year";
```

Now, linting flags that you've got an unnecessarily quoted identifier:

```
❯ sqlfluff lint --dialect postgres test.sql
== [test.sql] FAIL                                                                                                                                                        
L:   1 | P:  16 | RF06 | Unnecessary quoted identifier "year".
                       | [references.quoting]
All Finished 📜 🎉!
```

### Dialect

Postgres

### Version

```
❯ sqlfluff --version 
sqlfluff, version 3.2.4
```

### Configuration

No configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by craiga** [comment]
FWIW, I can work around this problem by adding the following to `.sqlfluff`:

```
[sqlfluff:rules:references.quoting]
ignore_words = year
```

Perhaps the solution to this problem is to have this `ignore_words` default to the list of reserved keywords?

**Comment by alanmcruickshank** [comment]
I don't think this one is a bug, it's working as expected, however I think for your project you should disable `RF04`. From the [docs](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-references.quoting):

> Although unreserved keywords can be used as identifiers, and reserved words can be used as quoted identifiers, best practice is to avoid where possible, to avoid any misunderstandings as to what the alias represents.

Based on what you've said in your issue, I think the practice in your team is to still use unreserved keywords, and not quote them (i.e. default behaviour of RF06, with RF04 disabled). If you'd prefer the unreserved keyword to be quoted, then set the `prefer_quoted_keywords` config for RF06 to `True`.
